main.py

```diff
@@ -0,0 +1,99 @@
+import streamlit as sl
+from transformers import pipeline, set_seed
+
+
+def is_int(str):
+    try:
+        int(str)
+        return True
+    except ValueError:
+        return False
+
+
+@sl.cache(suppress_st_warning=True, allow_output_mutation=True)
+def init():
+    return pipeline('text-generation', model='gpt2')
+
+
+GENERATOR = init()
+
+
+def generate_text(input_text, seed, max_len, num_return_seq):
+    if is_int(seed):
+        seed = abs(int(seed))
+        if seed > 2 ** 32 - 1:
+            set_seed(42)
+        else:
+            set_seed(seed)
+    else:
+        set_seed(42)
+
+    if is_int(max_len):
+        max_len = abs(int(max_len))
+    else:
+        max_len = 30
+
+    if is_int(num_return_seq):
+        num_return_seq = abs(int(num_return_seq))
+    else:
+        num_return_seq = 5
```

Comment:
This does not look very good. The behavior isn't obvious and the default values are hard-coded inside the code. There is no need for explicit type checking and you can specify default values in function arguments.